plugins/module_utils/github.py

```diff
@@ -1128,7 +1138,13 @@ def _manage_repository(self, state, current=None, check_mode=False, **kwargs):
         if current_repo and self._is_repo_update_needed(current_repo, kwargs):
             changed = True
             if not check_mode:
-                current_repo = self.update_repo(owner, repo_name, **kwargs)
+                new_kwargs = kwargs
+                # remove default_branch from payload as it would
+                # introduce branch handling for an empty repository
+                # causing the api to fail updating the repository.
+                if was_just_created:
```

Comment:
This approach will fail on second invocation of the module. Ideally we should check whether branch exist

Reply:
will it really? Since this should only be triggered, if the repository is not present. After creation, this would never be executed. Or am I missing something here?

Reply:
I guess it may happen if you don't specify branch protections (in this case auto_init is not triggered) and on next run you try to set default_branch, but it still doesn't exist

Reply:
@gtema can you take another look at this?

Reply:
I have had another look at this. Actually current integration tests show it is working already now (look at https://github.com/opentelekomcloud/ansible-collection-gitcontrol/blob/main/tests/integration/targets/github_org_repository/tasks/main.yaml) In the #41 I have also ensured this test is working and extended check to also try having auto_init without specifying repo templates